src/argilla/monitoring/asgi.py

```diff
@@ -113,18 +113,36 @@ def __init__(
         **kwargs
     ):
         super().__init__(*args, **kwargs)
+
         self._endpoint = api_endpoint
         self._dataset = dataset
         self._records_mapper = records_mapper or text_classification_mapper
+        self._queue = None
+        self._worker_task = None
+        self._initialized = False
+
+    def init(self):
+        if self._initialized:
+            return
+
         self._queue = Queue()
         self._worker_task = threading.Thread(
-            target=self.__worker__, name=argillaLogHTTPMiddleware.__name__, daemon=True
+            target=self.__worker__,
+            name=ArgillaLogHTTPMiddleware.__name__,
+            daemon=True,
         )
         self._worker_task.start()
+        atexit.register(self._worker_task.join)
+        self._initialized = True
 
     async def dispatch(
-        self, request: Request, call_next: RequestResponseEndpoint
+        self,
+        request: Request,
+        call_next: RequestResponseEndpoint,
     ) -> Response:
+        if not self._initialized:
+            self.init()
+
         if self._endpoint != request.url.path:  # Filtering endpoint path
             return await call_next(request)
 
```

Comment:
The Middleware is serializable but doesn´t allow for requests without a "Content-type" header. If it isn´t present `None` is returned. So the `if "application/json" not in content_type:`will result in an error. 

Additionally, I think we would also like to be able to log GET requests to endpoints that are processed correctly by the FastAPI. 

In both cases, I would say that the error handling must be fore explicit in terms of only letting the logging fail and not the entire requests.